tests/test_trainer.py

```diff
@@ -0,0 +1,69 @@
+import shutil
+import unittest
+from pathlib import Path
+
+from test_utils import TEST_CONFIG
+
+from trainlib.trainer import SegmentationTrainer
+
+
+class TestSegmentationTrainer(unittest.TestCase):
+    config = TEST_CONFIG
+
+    def test_in_order(self):
+        self._test_one_epoch()
+        self._test_folder_structure()
+
+    def _test_one_epoch(self):
+        trainer = SegmentationTrainer(config=self.config)
+        trainer.run()
+
+    def _test_folder_structure(self):
+        """Check that all folders for logging have been created"""
+        # check runs were created
+        self.assertPathExists("runs")
+        # check subdirs of runs (logs, output, trainlib, patch) were created
+        self.assertPathExists("runs/name/logs")
+        self.assertPathExists("runs/name/logs/train_logs.csv", is_file=True)
+
+        self.assertPathExists("runs/name/output")
+        self.assertPathExists("runs/name/output/preds")
+        self.assertPathExists("runs/name/output/preds/image.pt", is_file=True)
+        self.assertPathExists("runs/name/output/preds/label.pt", is_file=True)
+        self.assertPathExists("runs/name/output/preds/pred_epoch_1.pt", is_file=True)
+        for file in [
+            "emissions.csv",
+            "HausdorffDistance_raw.csv",
+            "MeanDice_raw.csv",
+            "metrics.csv",
+            "SurfaceDistance_raw.csv",
+            "val_mean_dice_raw.csv",
```

Comment:
These filenames are duplicated (mentioned again somewhere else in the codebase). So we could put them into a constants/utils file to be DRY

Reply:
The names ending with `_raw.csv` are generated automatically by the `MetricsSaver` (a MONAI handler). This is the first time I write them down. It's not optimal as this will fail as soon as the handler changes or the metric name changes (or I decide to use another keyx metric). But adding a configs/constants file would just increase the code footprint in this case.